tools/ibistool/ibistool.cpp

```diff
@@ -0,0 +1,430 @@
+  pm.addPass(mlir::createConvertSCFToCFPass());
+  pm.addPass(createSimpleCanonicalizerPass());
+}
+
+static void loadHandshakeTransformsPipeline(OpPassManager &pm) {
+  pm.addPass(circt::createCFToHandshakePass(
+      /*sourceConstants=*/false,
+      /*disableTaskPipelining=*/false));
+  pm.addPass(createSimpleCanonicalizerPass());
+  pm.nest<handshake::FuncOp>().addPass(
+      handshake::createHandshakeMaterializeForksSinksPass());
+  pm.addPass(createSimpleCanonicalizerPass());
+  pm.nest<handshake::FuncOp>().addPass(
+      handshake::createHandshakeInsertBuffersPass("all", 2));
+  pm.addPass(createSimpleCanonicalizerPass());
+}
+
+static void loadDCTransformsPipeline(OpPassManager &pm) {
+  pm.addPass(circt::createHandshakeToDCPass());
+  pm.addPass(createSimpleCanonicalizerPass());
+  pm.addPass(circt::createDCToHWPass());
+}
+
+static void loadESILoweringPipeline(OpPassManager &pm) {
+  pm.addPass(circt::esi::createESIPortLoweringPass());
+  pm.addPass(circt::esi::createESIPhysicalLoweringPass());
+  pm.addPass(circt::esi::createESItoHWPass());
+}
+
+static void loadHWLoweringPipeline(OpPassManager &pm) {
+  pm.addPass(createSimpleCanonicalizerPass());
+  pm.nest<hw::HWModuleOp>().addPass(circt::seq::createLowerSeqHLMemPass());
+  pm.addPass(sv::createHWMemSimImplPass(false, false));
+  pm.addPass(circt::createLowerSeqToSVPass());
+  pm.nest<hw::HWModuleOp>().addPass(sv::createHWCleanupPass());
+  pm.addPass(mlir::createCSEPass());
+
+  // Legalize unsupported operations within the modules.
+  pm.nest<hw::HWModuleOp>().addPass(sv::createHWLegalizeModulesPass());
+  pm.addPass(createSimpleCanonicalizerPass());
+
+  // Tidy up the IR to improve verilog emission quality.
+  auto &modulePM = pm.nest<hw::HWModuleOp>();
+  modulePM.addPass(sv::createPrettifyVerilogPass());
+}
+
+static void loadPipelineLoweringPipeline(OpPassManager &pm) {
+  pm.addPass(pipeline::createExplicitRegsPass());
+  pm.addPass(createPipelineToHWPass());
+}
+
+// --------------------------------------------------------------------------
+// Tool driver code
+// --------------------------------------------------------------------------
+
+static void loadLowLevelPassPipeline(
+    PassManager &pm, ModuleOp module,
+    std::optional<std::unique_ptr<llvm::ToolOutputFile>> &outputFile) {
+  loadPipelineLoweringPipeline(pm);
+  loadESILoweringPipeline(pm);
+  loadHWLoweringPipeline(pm);
+  if (traceIVerilog)
+    pm.addPass(circt::sv::createSVTraceIVerilogPass());
+
+  if (loweringOptions.getNumOccurrences())
+    loweringOptions.setAsAttribute(module);
+  if (outputFormat == OutputVerilog) {
+    pm.addPass(createExportVerilogPass((*outputFile)->os()));
+  } else if (outputFormat == OutputSplitVerilog) {
+    pm.addPass(createExportSplitVerilogPass(outputFilename));
+  }
+}
+
+static void loadIbisHiFlow(
+    PassManager &pm, ModuleOp module,
+    std::optional<std::unique_ptr<llvm::ToolOutputFile>> &outputFile) {
+  llvm::errs() << "Running high-level Ibis flow\n";
```

Comment:
Do you really want to unconditionally print this to err? Also, this message is misleading as it's not being "run"... it "will be run".

Reply:
Will nest under `  if (verbosePassExecutions)`